core/src/main/java/org/elasticsearch/index/analysis/CustomAnalyzerProvider.java

```diff
@@ -77,13 +78,20 @@ public void build(AnalysisService analysisService) {
             tokenFilters.add(tokenFilter);
         }
 
-        int positionOffsetGap = analyzerSettings.getAsInt("position_offset_gap", 0);
-        int offsetGap = analyzerSettings.getAsInt("offset_gap", -1);
+        int positionIncrementGap = 0;
+
+        if (Version.indexCreated(indexSettings).before(Version.V_2_0_0) ){
+            positionIncrementGap = analyzerSettings.getAsInt("position_offset_gap", 0);
+        }else{
```

Comment:
Users that have indexes created before 2.0 should not be precluded from using the new setting name. This looks like it *only* allows using the old name with indexes before 2.0.